components/places/src/bookmark_sync/record.rs

```diff
@@ -450,4 +497,74 @@ mod tests {
             })
         );
     }
+
+    // XXX - all below 'dateadded' tests only check the 'BookmarkItemRecord' variant.
+    // Is there a sane way to check all? Should we maybe have a specific type
+    // avoiding the need to remember `date_added: Option<i64>` needs the serde
+    // `deserialize_with` attribute?
+    fn check_date_added(j: serde_json::Value, expected: Option<i64>) {
+        let r: BookmarkItemRecord = serde_json::from_value(j).expect("should deserialize");
+        match &r {
+            BookmarkItemRecord::Bookmark(b) => assert_eq!(b.date_added, expected),
+            _ => panic!("unexpected record type"),
+        };
+    }
+
+    #[test]
+    fn test_dateadded_missing() {
+        check_date_added(
+            json!({"id": "unfiled", "parentid": "menu", "type": "bookmark"}),
+            None,
+        )
+    }
+
+    #[test]
+    fn test_dateadded_int() {
+        check_date_added(
+            json!({"id": "unfiled", "parentid": "menu", "type": "bookmark", "dateAdded": 123}),
+            Some(123),
+        )
+    }
+
+    #[test]
+    fn test_dateadded_negative() {
+        check_date_added(
+            json!({"id": "unfiled", "parentid": "menu", "type": "bookmark", "dateAdded": -1}),
+            Some(0),
+        )
+    }
+
+    #[test]
+    fn test_dateadded_str() {
+        check_date_added(
+            json!({"id": "unfiled", "parentid": "menu", "type": "bookmark", "dateAdded": "123"}),
+            Some(123),
+        )
+    }
+
+    // A kinda "policy" decision - like serde, 'type errors' fail rather than default.
```

Comment:
👍 If we see anything else in the wild, we'll have Sentry or engine failure telemetry tell us.